### PR TITLE
feat: add time unit to timestream write API

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix

- Feature

Current timestream write API is limited to MILLISECONDS precision only. This PR attempts to enable other time units (SECONDS, MICROSECONDS and NANOSECONDS).

One remaining issue is how to handle NANOSECONDS precision. `datetime` python objects are limited to microseconds precision so we would not be able to handle that.

### Detail
- Allow `time_unit` parameter to write method

### Relates
- #2261

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/test_timestream.py [hunk 1]
```diff
@@ -657,3 +657,40 @@ def test_batch_load(timestream_database_and_table, path, path2, time_unit, keep_
         """,
     )
     assert df2.shape == (len(df.index), len(df.columns) - 1)
+
+
+@pytest.mark.parametrize(
+    "time_unit",
```
Comment (kukushking) [inline tests/unit/test_timestream.py:663]: Minor - you can avoid having to refer by index `time_unit[0]` by de-structuring the parameter into two variables `time_unit,time_unit_exp` and pass the two as an input to the test case

**Review by malachi-constant — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: c8fe4f19dc8b8edd00a62849daf324c23b3b0bc2
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c937b04f-a5b6-4f7a-9729-2c2837aa0fd1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: c8fe4f19dc8b8edd00a62849daf324c23b3b0bc2
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c96da00f-4cc9-42f3-ad53-3f7c8c30f39b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 3264d09fa6f8a0231f75ab704a3ebaed86c06bb4
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e3fb80d7-0118-4b78-9d96-60241f4dc659%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 3264d09fa6f8a0231f75ab704a3ebaed86c06bb4
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d252c16a-e316-4b34-aa54-1530d9e6d0a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: tests/unit/test_timestream.py (hunk 1 above)
Comment (jaidisido) [reply tests/unit/test_timestream.py:663]: I don't think so, that would create combinations that are not correct (e.g. ("SECONDS", 3) and ("SECONDS", 6) when only ("SECONDS", 1) is correct)

Comment (kukushking) [reply tests/unit/test_timestream.py:663]: There is syntax specifically designed for parameter combinations, here's an example: https://github.com/aws/aws-sdk-pandas/blob/a1957c279afd6e66f1b9fcf3114e4068009cec64/tests/unit/test_athena.py#L323

File: awswrangler/timestream/_write.py [hunk 2]
```diff
@@ -39,36 +46,46 @@ def _df2list(df: pd.DataFrame) -> List[List[Any]]:
...
     # Values in common_attributes take precedence
     common_attributes.setdefault("Version", version)
+    common_attributes.setdefault("TimeUnit", _check_time_unit(common_attributes.get("TimeUnit", time_unit)))
 
-    if "Time" not in common_attributes:
-        # TimeUnit is MILLISECONDS by default for Timestream writes
-        # But if a time_col is supplied (i.e. Time is not in common_attributes)
-        # then TimeUnit must be set to MILLISECONDS explicitly
-        common_attributes["TimeUnit"] = "MILLISECONDS"
+    if "Time" not in common_attributes and common_attributes["TimeUnit"] == "NANOSECONDS":
```
Comment (jaidisido) [inline awswrangler/timestream/_write.py:87]: Python datetime only supports up to microseconds

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 02ef70403d55f1830d882fbf079522a877bda730
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ad5b3993-4ead-4f51-88bd-90d38351f6b1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 02ef70403d55f1830d882fbf079522a877bda730
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1689e53a-ea25-4d97-9795-d83d4e3f76de%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: d401ec82fc03fafb840c90b9414e35cba4070094
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=58d1a4f4-4522-434f-8e68-a8c97d6bc96b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: d401ec82fc03fafb840c90b9414e35cba4070094
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=19f10b5e-1173-4fa7-bff6-1a635b804304%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**